sashimi/hardware/scanning/scanloops.py

```diff
@@ -258,6 +259,108 @@ def fill_arrays(self):
         self.wait_signal.clear()
 
 
+class TriggeredPlanarScanLoop(ScanLoop):
+    """Class for controlling the planar scanning mode, where we image only one plane and
+    do not control the piezo and vertical galvo."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        buffer_len = int(round(self.sample_rate / self.parameters.triggering.frequency))
+        self.camera_pulses = RollingBuffer(buffer_len)
+        set_impulses(
+            self.camera_pulses.buffer,
+            1,
+            n_skip_start=0,
+            n_skip_end=0,
+        )
+        self.current_frequency = self.parameters.triggering.frequency
+        self.camera_on = False
+        self.trigger_exp_start = False
+        self.camera_was_off = True
+        self.wait_signal.set()
+
+    def initialize(self):
+        super().initialize()
+        self.camera_on = False
+        self.wait_signal.set()
+
+    def loop_condition(self):
+        return (
+            super().loop_condition() and self.parameters.state == ScanningState.TRIGGERED_PLANAR
+        )
+
+    def n_samples_period(self):
+        if (
+            self.parameters.triggering.frequency is None
+            or self.parameters.triggering.frequency == 0
+        ):
+            return super().n_samples_period()
+        else:
+            n_samples_trigger = int(
+                round(self.sample_rate / self.parameters.triggering.frequency)
+            )
+            return max(n_samples_trigger, super().n_samples_period())
```

Comment:
Admittedly, I don't fully understand what is happening here. In the triggerless planar mode this used to be encapsulated by the least common multiple. I changed it to `max()` like in the `VolumetricScanLoop` as it made sense to me that it would be the size of the frame trigger plus the time until the next one.